### ISSUE TITLE
Logout is not working

### ISSUE THREAD

**Issue by RomaSV** [RomaSV/HotelsWebService#1]
Как сделать logout в данном проекте? Я понимаю, что по-хорошему нужно делать POST-запрос, но пока нет клиентской части, это проблематично тестировать.
Нашел такое решение:
`.logout().logoutRequestMatcher(new AntPathRequestMatcher("/logout"))`
Но оно не работает. Это где то я косячу, или оно в данном случае не должно работать?
И все таки, POST, или пока GET сойдет?

**Comment by JohnKhandygo** [comment]
Для debug подойдет swagger. http://www.baeldung.com/swagger-2-documentation-for-spring-rest-api

**Comment by RomaSV** [comment]
Добавил swagger. Не могу понять, почему при GET запросе (который только для менеджера) через логин менеджера все работает (как и должно быть), НО при POST или PUT запросах, которые настроены так же, для менеджера кидает 403 (а должно пропускать).
С чем это может быть связано?

Например: 
- /hotels (GET) - работает
 - /hotels?adminId=1 (POST) - не пускает

**Comment by RomaSV** [comment]
Пытаюсь разобраться с feign (openfeign), и что-то плохо выходит.
Как я понял, достаточно чего-то вроде этого:
```
public class HotelFeignClientBuilder {
    private  HotelService hotelService = Feign.builder()
            .decoder(new GsonDecoder())
            .target(HotelService.class, "http://localhost:8080");
    public HotelService getHotelService() {
        return hotelService;
    }
}
```
Но при попытке использовать такой hotelService (сервер запущен) кидает NPE.
Вообще не могу понять, эта версия feign умеет работать с RequestMapping? Или надо для него отдельно такой же интерфейс, но с RequestLine писать? Последнее выглядит немного странно, учитывая, что интерфейсы одинаковы в остальном

**Comment by JohnKhandygo** [comment]
Да, Вам необходимо будет написать отдельный интерфейс с `RequestLine`. Есть вариант автоматической генерации API, но это достаточно сложно в maven настраивается, поэтому на свой страх и риск, если хотите этим путем идти.

**Comment by JohnKhandygo** [comment]
Скиньте полный stacktrace. Либо уже завтра на паре посмотрим. У меня с небольшими исправлениями завелось на той версии кода, которая сейчас в репозитории.

**Comment by RomaSV** [comment]
Уже прогресс есть, почти все ок. Но он спотыкается об команты:
`Expected BEGIN_ARRAY but was BEGIN_OBJECT at line 1 column 149 path $[0].rooms[0]`
При том, что в комнате, которая передается в запросах, нет ни массивов, ни коллекций.
Если не таскать комнаты вместе с отелем, то все работает. Пока не могу понять, почему так

P.S. Запушил, но там пока что только `getHotels` ипользует новые `HotelData` и `RoomData`.

**Comment by JohnKhandygo** [comment]
Понял. Я постараюсь посмотреть завтра.

**Comment by JohnKhandygo** [comment]
Прошу прощения, удалось только сейчас посмотреть. Ошибка заключается в несоответствии методов `getHotels` в интерфейсе `HotelsClient` и в контроллере `HotelResource`.

**Comment by RomaSV** [comment]
Увидел. Глупо... 
Как то пропустил, не заметил. Но хорошо, это быстро исправить. Мне для юзеров делать отдельного клиента, или надо их объединять в один (юзеры и отели в разных контроллерах)?

**Comment by JohnKhandygo** [comment]
На каждый spring controller обычно заводят свой API, если Вы об этом.

**Comment by JohnKhandygo** [comment]
Отлично, Вы большой молодец, что освоили все эти штуки, полагаю, что они Вам пригодятся в будущем. Зачтено на 5, за курсовой соответственно тоже 5. 

Проект, конечно, не идеальный, но, учитывая, что Вы осваивали все сами, это очень круто. Если хотите, я могу попробовать выкатить набор более менее формальных замечаний, но уже после сессии скорее всего.

**Comment by RomaSV** [comment]
Хорошо, спасибо! По поводу замечаний было бы замечательно

**Comment by JohnKhandygo** [comment]
Окей, я постараюсь не забыть.
